src/jdom/servicehost.ts

```diff
@@ -0,0 +1,76 @@
+import { BaseReg, SystemCmd, SystemEvent } from "../../jacdac-spec/dist/specconstants";
+import { NumberFormat, setNumber } from "./buffer";
+import JDDeviceHost from "./devicehost";
+import { JDEventSource } from "./eventsource";
+import Packet from "./packet";
+import JDRegisterHost from "./registerhost";
+import { memcpy } from "./utils";
+
+export default class JDServiceHost extends JDEventSource {
+    public serviceIndex: number = -1; // set by device
+    public device: JDDeviceHost;
+    private readonly _registers: JDRegisterHost[] = [];
+    private readonly commands: { [identifier: number]: (pkt) => void } = {};
+
+    constructor(public readonly serviceClass: number) {
+        super();
+
+        this.addRegister(BaseReg.StatusCode, "u16 u16", [0, 0])
+    }
+
+    get registers() {
+        return this._registers.slice(0);
+    }
+
+    register(identifier: number) {
+        return this._registers.find(reg => reg.identifier === identifier);
+    }
+
+    get statusCode() {
+        return this._registers.find(reg => reg.identifier === BaseReg.StatusCode);
+    }
+
+    protected addRegister(identifier: number, packFormat: string, defaultValue: any[]) {
+        const reg = new JDRegisterHost(this, identifier, packFormat, defaultValue);
+        this._registers.push(reg);
+        return reg;
+    }
+
+    protected addCommand(identifier: number, handler: (pkt) => void) {
```

Comment:
type missing on `pkt` ? we should have `noImplicitAny` in tsconfig

Reply:
need to find this option in gatsby

Reply:
you can just add it to main tsconfig.json

```
  "noImplicitAny": true,
    "noImplicitReturns": true,
```

There is probably a few places to fix